client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/use-site-migration-credentials-mutation.tsx

```diff
@@ -0,0 +1,56 @@
+import { useMutation } from '@tanstack/react-query';
+import { useCallback } from 'react';
+import { useSiteSlugParam } from 'calypso/landing/stepper/hooks/use-site-slug-param';
+import wp from 'calypso/lib/wp';
+import { CredentialsFormData } from './types';
+
+export const useMigrationCredentialsMutation = () => {
+	const siteSlug = useSiteSlugParam();
+
+	const saveCredentialsMutation = useMutation( {
+		mutationFn: async ( {
+			siteAddress,
+			username,
+			password,
+			notes,
+			howToAccessSite,
+		}: CredentialsFormData ) => {
+			try {
+				await wp.req.post( {
+					path: `/help/automated-migration`,
+					apiNamespace: 'wpcom/v2',
+					body: {
+						blog_url: siteSlug,
+						from_url: siteAddress,
+						migration_type: howToAccessSite,
+						username,
+						password,
+						notes,
+					},
+				} );
+			} catch ( error: any ) {
+				throw {
+					body: {
+						code: error.code,
+						message: error.message,
+						data: error.data,
+					},
+					status: error.status,
+				};
+			}
```

Comment:
I think we should also not be handling the error here, and should try to keep a simple request and workout the responses on the API:
```suggestion
			let body: AutomatedMigrationBody = {
				migration_type: howToAccessSite,
				notes,
			};

			if ( howToAccessSite === 'credentials' ) {
				body = {
					...body,
					from_url: siteAddress,
					username,
					password,
				};
			} else {
				// In case of backup, we need to send the backup file location
				body = {
					...body,
					from_url: backupFileLocation,
				};
			}

			return wpcomRequest( {
				path: 'help/automated-migration',
				apiNamespace: 'wpcom/v2/',
				apiVersion: '2',
				method: 'POST',
				body,
			} );
```

Reply:
I've removed the try catch and not handling the error here anymore. Just for now till we handle it in API part, I have extracted only the mapping part to a separate function to make it keep working without any change. We can just remove this function later if we unify the error body structure later from the API.